lib/middleware/send-message/user-validate.js

```
@@ -0,0 +1,34 @@
+'use strict';
+
+const helpers = require('../../helpers');
+const UnprocessibleEntityError = require('../../../app/exceptions/UnprocessibleEntityError');
+
+module.exports = function validateUser() {
+  return (req, res, next) => {
+    if (!req.user) {
+      return next();
+    }
+
+    if (req.user.sms_status === helpers.subscriptionStatusStopValue()) {
+      const error = new UnprocessibleEntityError('Northstar User is unsubscribed.');
+      return helpers.sendErrorResponse(res, error);
+    }
+
+    if (req.user.sms_paused) {
+      const error = new UnprocessibleEntityError('Northstar User conversation is paused.');
+      return helpers.sendErrorResponse(res, error);
+    }
+
+    try {
+      /**
+       * Sanity check E.164 format.
+       */
+      req.platformUserId = helpers.formatMobileNumber(req.user.mobile);
+    } catch (err) {
+      const error = new UnprocessibleEntityError('Cannot format Northstar User mobile.');
```

I think we need to suppress errors here for messages to users with no valid mobile #.

---

@sergii-tkachenko -- I thought Blink always suppresses 422's, is that not the case? Could make sense to leave the explicitly suppress, but there could also be that (maybe very slight?) chance that we fix the user number before the retry count reaches the max...

---

Mmm, maybe. This was a 500 error so I just suppressed when @sergii-tkachenko reported that it should be suppressed. I wasn't quite sure about Blink suppressing all 422s.

---

It always suppresses 422s. They show up in logs as warnings though.

---

You don't need to explicitly suppress 422s, it is a part of our retry contract. We need to write it down somewhere.